controllers/discovery/initDiscoveryUserData.js

```diff
@@ -9,11 +9,17 @@ const {QueryTypes} = require('sequelize');
  * @returns
  */
 const InitDiscoveryUserData = async (req, res) => {
-  const {limit = 50, page = 0} = req.body;
+  const {limit = 50, page = 0} = req.query;
 
   const {userId} = req;
 
   try {
+    let totalDataQuery = `
+    SELECT 
+        count(A.user_id) as total_data
+    FROM users A
+    WHERE A.user_id != :userId AND A.is_anonymous = false AND A.is_banned = false`;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The SQL query used to count the total number of users does not consider the possibility of SQL injection attacks. Although Sequelize escapes replacements by default, it's still a good practice to use Sequelize's built-in methods for creating queries whenever possible. This can help prevent potential security vulnerabilities.



<!-- This is an auto-generated comment by OSS CodeRabbit -->